llvm/lib/IR/LLVMContext.cpp

```diff
@@ -120,6 +120,12 @@ void LLVMContext::removeModule(Module *M) {
   pImpl->OwnedModules.erase(M);
 }
 
+unsigned LLVMContext::generateMachineFunctionNum(Function &F) {
+  Module *M = F.getParent();
+  assert(pImpl->OwnedModules.contains(M) && "Unexpected module!");
+  return pImpl->MachineFunctionNums[M]++;
```

Comment:
do these need to start at 0 per module? otherwise could we just keep counting up ignoring the module?

otherwise we'll have to erase these entries in `removeModule()` or risk stale entries getting reused